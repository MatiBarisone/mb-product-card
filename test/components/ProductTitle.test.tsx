import React from "react";
import { render } from "@testing-library/react";
import { ProductTitle, ProductCard } from '../../src/components';
import { product1 } from './data/products';


describe('ProductTitle', () => {
    test('debe mostrar el componente correctamente con el tÃ­tulo personalizado', () => {
        const { asFragment } = render(
            <ProductTitle title="Custom Product" className="custom-class" />
        );
    
        expect(asFragment()).toMatchSnapshot();
    });



    test('debe de mostrar el componente con el nombre del producto', () => {
        const { asFragment } = render(
            <ProductCard product={product1}>
                {() => <ProductTitle />}
            </ProductCard>
        );
    
        expect(asFragment()).toMatchSnapshot();
    });

});