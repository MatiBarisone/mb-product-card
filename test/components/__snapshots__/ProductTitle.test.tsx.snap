// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle debe de mostrar el componente con el nombre del producto 1`] = `
<DocumentFragment>
  <div
    class="productCard undefined"
  >
    <span
      class="productDescription undefined"
    >
      Coffee Mug - Sin imagen
    </span>
  </div>
</DocumentFragment>
`;

exports[`ProductTitle debe mostrar el componente correctamente con el t√≠tulo personalizado 1`] = `
<DocumentFragment>
  <span
    class="productDescription custom-class"
  >
    Custom Product
  </span>
</DocumentFragment>
`;
